### PR TITLE
enh(ci): promote testing package

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: .github/workflows/get-version.yml [hunk 1]
```diff
@@ -94,7 +94,7 @@ jobs:
             release* | hotfix*)
               STABILITY="testing"
               ;;
-            master | [2-9][0-9].[0-9][0-9].x)
+            master | [2-9][0-9].[0-9][0-9].x | MON-16968-update-workflows-to-make-release-to-stable-faster )
```
Comment (tuntoja) [inline .github/workflows/get-version.yml:97]:
```suggestion
            master | [2-9][0-9].[0-9][0-9].x)
```

File: .github/actions/promote-to-stable/action.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      if: ${{ startsWith(inputs.distrib, 'el') }}
+      run: |
+        set -x
+        echo "[DEBUG] - Major version: ${{ inputs.major_version }}"
+        echo "[DEBUG] - Minor version: ${{ inputs.minor_version }}"
+        echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
+
+        for ARCH in "noarch" "x86_64"; do
+          echo "[DEBUG] - Get path of $ARCH testing artifacts to promote to stable."
+          SRC_PATHS+=$(jf rt s --include-dirs rpm-standard/${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module }}/*.rpm | jq -r '.[].path')
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:41]: Shouldn't SRC_PATH be set before being affected to this operation ?

**Review by sc979 — APPROVED** [review]

File: .github/actions/promote-to-stable/action.yml (hunk 2 above)
Comment (tuntoja) [reply .github/actions/promote-to-stable/action.yml:41]: Added initialization before first loop

**Review by bdauria — DISMISSED** [review]

File: .github/workflows/web.yml [hunk 3]
```diff
@@ -130,7 +136,7 @@ jobs:
 
   frontend-lint:
     runs-on: ubuntu-22.04
-    needs: [changes]
+    needs: [changes, get-version]
```
Comment (kduret) [inline .github/workflows/web.yml:139]:
```suggestion
    needs: [changes]
```

File: .github/workflows/web.yml [hunk 4]
```diff
@@ -143,7 +149,7 @@ jobs:
 
   frontend-unit-test:
     runs-on: ubuntu-22.04
-    needs: [changes]
+    needs: [changes, get-version]
```
Comment (kduret) [inline .github/workflows/web.yml:152]:
```suggestion
    needs: [changes]
```

File: .github/workflows/web.yml [hunk 5]
```diff
@@ -174,7 +180,7 @@ jobs:
           path: "centreon/junit.xml"
 
   frontend-component-test:
-    needs: [changes]
+    needs: [changes, get-version]
```
Comment (kduret) [inline .github/workflows/web.yml:183]:
```suggestion
    needs: [changes]
```

File: .github/workflows/web.yml [hunk 6]
```diff
@@ -184,7 +190,7 @@ jobs:
 
   backend-unit-test:
     runs-on: ubuntu-22.04
-    needs: [changes]
+    needs: [changes, get-version]
```
Comment (kduret) [inline .github/workflows/web.yml:193]:
```suggestion
    needs: [changes]
```

File: .github/workflows/web.yml [hunk 7]
```diff
@@ -268,8 +274,8 @@ jobs:
 
   dockerize:
     runs-on: ubuntu-22.04
-    needs: [get-version, package]
-    if: ${{ !cancelled() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled') }}
+    needs: [package, get-version]
```
Comment (kduret) [inline .github/workflows/web.yml:277]:
```suggestion
    needs: [get-version, package]
```

File: .github/workflows/awie.yml [hunk 8]
```diff
@@ -32,6 +32,8 @@ jobs:
   package:
     needs: [get-version]
 
+    if: ${{ contains(fromJson('["testing", "unstable"]'), needs.get-version.outputs.stability) }}
```
Comment (kduret) [inline .github/workflows/awie.yml:35]: to apply everywhere :)
```suggestion
    if: ${{ needs.get-version.outputs.stability != 'stable' }}
```

**Review by kduret — APPROVED** [review]

File: .github/actions/promote-to-stable/action.yml [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+    - name: Promote RPM packages to stable
+      if: ${{ startsWith(inputs.distrib, 'el') }}
+      run: |
+        set -x
+        echo "[DEBUG] - Major version: ${{ inputs.major_version }}"
+        echo "[DEBUG] - Minor version: ${{ inputs.minor_version }}"
+        echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
+
+        SRC_PATHS=""
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:39]:
```suggestion
```
Not needed anymore. The affectation will be done in line 43

File: .github/actions/promote-to-stable/action.yml [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+
+        for ARCH in "noarch" "x86_64"; do
+          echo "[DEBUG] - Get path of $ARCH testing artifacts to promote to stable."
+          SRC_PATHS=$(jf rt s --include-dirs rpm-standard/${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module }}/*.rpm | jq -r '.[].path')
+
+          for SRC_PATH in $SRC_PATHS;do
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:45]:
```suggestion
          for SRC_PATH in $SRC_PATHS; do
```

Comment (sc979) [reply .github/actions/promote-to-stable/action.yml:45]: shouldn't you call it as an array : $SRC_PATHS[@] ?

File: .github/actions/promote-to-stable/action.yml [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+        for ARCH in "noarch" "x86_64"; do
+          echo "[DEBUG] - Get path of $ARCH testing artifacts to promote to stable."
+          SRC_PATHS=$(jf rt s --include-dirs rpm-standard/${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module }}/*.rpm | jq -r '.[].path')
+
+          for SRC_PATH in $SRC_PATHS; do
+            echo "[DEBUG] - Source path found: $SRC_PATHS"
+          done
+
+          if [[ -z "SRC_PATHS" ]]; then
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:49]:
```suggestion
          if [[ -z "SRC_PATH" ]]; then
```

**Review by sc979 — COMMENTED** [review]
DEB promotion to be refactored.

File: .github/actions/promote-to-stable/action.yml [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+        echo "[DEBUG] - Minor version: ${{ inputs.minor_version }}"
+        echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
+
+        SRC_PATHS=""
+
+        for ARCH in "noarch" "x86_64"; do
+          echo "[DEBUG] - Get path of $ARCH testing artifacts to promote to stable."
+          SRC_PATHS=$(jf rt s --include-dirs rpm-standard/${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module }}/*.rpm | jq -r '.[].path')
+
+          for SRC_PATH in $SRC_PATHS; do
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:45]: shouldn't you call it as an array by using $SRC_PATHS[@] ?

File: .github/actions/promote-to-stable/action.yml [hunk 13]
```diff
@@ -71,10 +71,14 @@ runs:
         echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
 
         echo "[DEBUG] - Get path of testing DEB packages to promote to stable."
-        SRC_PATHS+=$(jf rt s --include-dirs apt-standard-${{ inputs.major_version }}-testing/pool/${{ inputs.module }}/*.deb | jq -r '.[].path')
-        echo "[DEBUG] - Source path: $SRC_PATHS"
+        SRC_PATHS=$(jf rt s --include-dirs apt-standard-${{ inputs.major_version }}-testing/pool/${{ inputs.module }}/*.deb | jq -r '.[].path')
 
-        if [[ -z "SRC_PATHS" ]]; then
+        if [[ ${SRC_PATHS[@]} ]]; then
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:76]: The value of the first key could be null.

**Review by sc979 — DISMISSED** [review]

File: .github/workflows/get-version.yml [hunk 14]
```diff
@@ -94,7 +94,7 @@ jobs:
             release* | hotfix*)
               STABILITY="testing"
               ;;
-            master | [2-9][0-9].[0-9][0-9].x)
+            master | [2-9][0-9].[0-9][0-9].x | MON-16968-update-workflows-to-make-release-to-stable-faster)
```
Comment (tuntoja) [inline .github/workflows/get-version.yml:97]:
```suggestion
            master | [2-9][0-9].[0-9][0-9].x)
```

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**